src/routes/PageLayout.tsx

```diff
@@ -0,0 +1,23 @@
+import React, { useEffect } from 'react'
+import { Outlet, useNavigate, useParams } from 'react-router-dom'
+import classes from '../App.module.css'
+
+function PageLayout() {
+    const { store } = useParams()
+    const navigate = useNavigate()
+
+    useEffect(() => {
+        const storeOptions = ['dataStore', 'userDataStore']
+        if (!storeOptions.includes(store)) {
+            navigate('dataStore')
+        }
+    }, [store])
```

Comment:
It might be a little bit better to do this in the render loop instead of in a `useEffect` - rendering `<Navigate>` if the specified store is invalid.  With a `useEffect` this will first render the page BEFORE redirecting.